dex/pair/src/safe_price_view.rs

```diff
@@ -461,9 +497,14 @@ pub trait SafePriceViewModule:
         let lp_supply_sum = BigUint::from(left_weight) * left_observation.lp_supply_accumulated
             + BigUint::from(right_weight) * right_observation.lp_supply_accumulated;
 
+        let timestamp_sum = BigUint::from(left_weight) * left_observation.recording_timestamp
+            + BigUint::from(right_weight) * right_observation.recording_timestamp;
+        let interpolated_timestamp = (timestamp_sum / weight_sum).to_u64().unwrap_or_default();
+
```

Comment:
Wasn't there a math module that did all the linear interpolation for you? This also applies to the lines above.

Reply:
@psorinionut

Reply:
Added weighted average from math module.